### PR TITLE
Partial fix #174 

### LINKED ISSUE #174: Speed up `draw_xgi_nodes` and `draw_xgi_hyperedges`
If we remove the for-loop in the function that draws the hypergraph nodes with `scatter` and instead pass iterables into the scatter function, it could be faster. Also, if we plot Collection objects instead of Patch objects, maybe it will be faster as well. (https://stackoverflow.com/questions/37573314/matplotlib-efficient-way-to-create-large-number-of-patch-objects)

### PR DESCRIPTION
**Pull request by iaciac**
Nodes are now plotted with iterables. 
The plotting of hyperedges has to be improved, but it is not as straightforward.

### CODE REVIEW

File: xgi/drawing/xgi_pylab.py [hunk 1]
```diff
@@ -206,17 +206,15 @@ def draw_xgi_nodes(
...
-        )
+    ax.scatter(
+        x=[pos[i][0] for i in H.nodes],
+        y=[pos[i][1] for i in H.nodes],
+        s=[node_size[i] ** 2 for i in H.nodes],
+        c=[node_fc[i] for i in H.nodes],
+        edgecolors=[node_ec[i] for i in H.nodes], 
+        linewidths=[node_lw[i] for i in H.nodes],
+        zorder=zorder,
+    )
```
Comment (leotrs) [inline xgi/drawing/xgi_pylab.py:217]: This is even faster as it only iterates over the nodes once:
```suggestion
    x, y, s, c, ec, lw, = zip(*[(pos[i][0], pos[i][1], node_size[i], node_fc[i], node_ec[i], node_lw[i]) for i in H.nodes])
    ax.scatter(x=x, y=y, s=s, c=c, edgecolors=ec, linewidths=lw, zorder=zorder)
```
(Or, uh, something like this, I haven't tested it!)

**Review by leotrs — COMMENTED** [review]
Thanks for PRing! Just a quick suggestion.

File: xgi/drawing/xgi_pylab.py [hunk 2]
```diff
@@ -206,17 +206,8 @@ def draw_xgi_nodes(
...
-        ax.scatter(
-            x,
-            y,
-            s=node_size[i] ** 2,
-            c=node_fc[i],
-            edgecolors=node_ec[i],
-            linewidths=node_lw[i],
-            zorder=zorder,
-        )
+    x, y, s, c, ec, lw, = zip(*[(pos[i][0], pos[i][1], node_size[i], node_fc[i], node_ec[i], node_lw[i]) for i in H.nodes])
```
Comment (iaciac) [inline xgi/drawing/xgi_pylab.py:209]:
```suggestion
    x, y, s, c, ec, lw, = zip(*[(pos[i][0], pos[i][1], node_size[i]**2, node_fc[i], node_ec[i], node_lw[i]) for i in H.nodes])
```

**Review by iaciac — COMMENTED** [review]
Thanks @leotrs, just the node size adjustment was missing

**Review by leotrs — APPROVED** [review]

**Comment by maximelucas** [comment]
Thank Iac. Just put a commented related to #181 but already approved.

**Review by maximelucas — APPROVED** [review]

**Comment by nwlandry** [comment]
@iaciac can you give a comparison on the speed of drawing nodes before and after your PR?

**Comment by iaciac** [comment]
If I remember correctly it's 3 times faster. I need to check

File: tutorials/Tutorial 5 - Plotting.ipynb [hunk 3]
```diff
@@ -108,13 +108,13 @@
     "cmap = plt.cm.Paired\n",
```
[the remark this answers is not in the document]
Comment (nwlandry) [reply tutorials/Tutorial 5 - Plotting.ipynb:108]: Okay, I looked into this and fixed it. The reason why it still runs (the incorrect args simply do nothing) is because we have **kwargs as an argument to update the default drawing settings (it was too wordy to write each setting out) and so it can accept any kwarg. If it doesn't mean anything, however, nothing will happen.
